ext/cosmos/ext/structure/structure.c

```diff
@@ -1417,7 +1413,16 @@ void Init_structure (void)
   rb_define_singleton_method(cBinaryAccessor, "write", binary_accessor_write, 7);
 
   cStructure = rb_define_class_under(mCosmos, "Structure", rb_cObject);
-  rb_const_set(cStructure, id_const_ZERO_STRING, ZERO_STRING);
+  id_const_ZERO_STRING = rb_intern("ZERO_STRING");
+  zero_string = rb_str_new((char*) &zero, 1);
+  rb_funcall(zero_string, id_method_freeze, 0);
+  rb_const_set(cStructure, id_const_ZERO_STRING, zero_string);
+  ZERO_STRING = rb_const_get(cStructure, id_const_ZERO_STRING);
+  id_const_ASCII_8BIT_STRING = rb_intern("ASCII_8BIT_STRING");
+  ascii_8bit_string = rb_str_new2("ASCII-8BIT");
+  rb_funcall(ascii_8bit_string, id_method_freeze, 0);
+  rb_const_set(cStructure, id_const_ASCII_8BIT_STRING, ascii_8bit_string);
+  ASCII_8BIT_STRING = rb_const_get(cStructure, id_const_ASCII_8BIT_STRING);
```

Comment:
This block is quite different. It makes sense to me in that you're setting the constant and then getting it to assign it to the variable. The old code never appears to set the constant and get it back. How did it work before?